snd_mem.c

```diff
@@ -669,10 +669,11 @@ sf_count_t SFVIO_Tell(void *user_data)
 static qbool S_ParseCueMark(const byte* chunk, int len, int cue_point_id, int* sample_length)
 {
 	int pos = 0;
+	unsigned int size = 1;
 
 	*sample_length = 0;
-	while (pos < len - 8) {
-		unsigned int size = BuffLittleLong(chunk + pos + 4);
+	while (pos < len - 8 && size !=0 ) {
```

Comment:
Can you add some braces around the logical operations, please?

Also, there should be a space before 0, and nospace before the closing brace ;)